src/config.ts

```diff
@@ -138,35 +143,47 @@ export function getStampsConfig({
   POSTAGE_TTL_MIN,
   POSTAGE_REFRESH_PERIOD,
   POSTAGE_EXTENDSTTL,
+  POSTAGE_EXTENDS_CAPACITY,
 }: EnvironmentVariables = {}): StampsConfig | undefined {
   const refreshPeriod = Number(POSTAGE_REFRESH_PERIOD || DEFAULT_POSTAGE_REFRESH_PERIOD)
   const beeDebugApiUrl = BEE_DEBUG_API_URL || DEFAULT_BEE_DEBUG_API_URL
 
   // Start in hardcoded mode
   if (POSTAGE_STAMP) return { mode: 'hardcoded', stamp: POSTAGE_STAMP }
   // Start autobuy
-  else if (!POSTAGE_EXTENDSTTL && POSTAGE_DEPTH && POSTAGE_AMOUNT && BEE_DEBUG_API_URL) {
+  else if (
+    (POSTAGE_EXTENDSTTL === 'true' &&
+      POSTAGE_AMOUNT &&
+      POSTAGE_DEPTH &&
+      Number(POSTAGE_TTL_MIN) >= MINIMAL_EXTENDS_TTL_VALUE) ||
+    POSTAGE_EXTENDS_CAPACITY === 'true'
```

Comment:
So I don't think this condition is really correct. IMHO it should be `POSTAGE_EXTENDSTTL === 'true' || POSTAGE_EXTENDS_CAPACITY === 'true'`, because these flags should enable the extends mode. Then inside the if branch, you should verify that all the required options are set and valid. I see that you have removed the functions that we introduced in the previous PR: `createAutoBuyStampsConfig()` etc. I would propose to bring those back and actually maybe removing the utility functions that you have introduced here (eq. `getAmount()` etc.) and move the parsing **and validation** of these values into the `create*()` functions. If you want, you can refactor the utility functions like `getAmount()` into more generic parsing functions like `getInteger()` etc. For inspiration (you can even copy&paste those) you can see what we do in [bee-js here](https://github.com/ethersphere/bee-js/blob/master/src/utils/type.ts#L87). We have in the pipeline refactoring these utility functions into some utility NPM package that we could use across our projects, but we have still not gotten around to doing that 🙈